File: TypeScript/Slice/remove-element-from-slice.ts
function main(): void {
    // Create a slice of strings
    var randomSlice: string[] = ["a", "b", "c", "d", "e"]
    // Remove an element from the slice, print the slice
    console.log(removeElementFromSlice(randomSlice, "d"))
}

main()

// Remove an element from the provided slice and return the new slice.
function removeElementFromSlice(originalSlice: string[], element: string): string[] {
    var index: number = originalSlice.indexOf(element)
    if (index > -1) {
        originalSlice.splice(index, 1)
    }
    return originalSlice
}

File: Dart/Slice/remove_element_from_slice.dart
import "dart:core";

void main() {
  // Create a new list of strings.
  final List<String> randomSlice = <String>["Bob", "Alice", "Joe"];
  // Remove an element from the slice.
  print(removeElementFromSlice(randomSlice, "Joe"));
}

// Remove an element from an slice.
List<dynamic> removeElementFromSlice(
  final List<dynamic> slice,
  final dynamic element,
) {
  slice.remove(element);
  return slice;
}
